LegiScan API documentation: https://legiscan.com/documentation/legiscan

In [1]:
import os
import time
import requests
import pandas as pd
from datetime import date
from dotenv import load_dotenv

load_dotenv() 
print ("Libraries imported")

Libraries imported


In [2]:
API_KEY = os.getenv("LEGI_API_KEY")

In [3]:
url = "https://api.legiscan.com/"
## I need to get a access code from the getDatasetList url used to call the GetDataset url.

In [4]:
params = {
    "key": API_KEY,
    "op": "getDatasetList",
    "state": "US"
}

response = requests.get(url, params=params)

print(response.status_code)

data = response.json()
print(data.keys())


200
dict_keys(['status', 'datasetlist'])


In [5]:
status = data["status"]
print(status)

OK


In [16]:
print(data["datasetlist"][0])


{'state_id': 52, 'session_id': 2199, 'year_start': 2025, 'year_end': 2026, 'prefile': 0, 'sine_die': 0, 'prior': 0, 'special': 0, 'session_tag': 'Regular Session', 'session_title': '2025-2026 Regular Session', 'session_name': '119th Congress', 'dataset_date': '2026-09-13', 'dataset_hash': '9262609d93844236c2ade4406c12a619', 'dataset_size': 63172657, 'dataset_size_csv': 4417424, 'access_key': '5IGESFEerbQQEAulIHxVJR'}


9/11/2026: <br>
    session_id: 2199 <br>
    access_key: '7GNHYv7EcN1RPXURARflAn' <br>
    dataset_hash: '99dcea213d48715d5dc603e1a48da9ac' <br>

9/17/2026: <br>
access_key: key = 5IGESFEerbQQEAulIHxVJR <br>
dataset_hash: 9262609d93844236c2ade4406c12a619 

In [17]:
print(data["datasetlist"][0]["dataset_hash"])

9262609d93844236c2ade4406c12a619


In [18]:
key = data["datasetlist"][0]["access_key"]
print(key)

5IGESFEerbQQEAulIHxVJR


In [19]:
session_id = data["datasetlist"][0]["session_id"]
print(session_id)

2199


In [20]:
params2= {
    "key": API_KEY,
    "op": "getDatasetRaw",
    "id": session_id,
    "access_key": key
}

response2 = requests.get(url, params=params2)

In [21]:
print(response2.status_code)

data2 = len(response2.content)
print(data2)

200
63172657


In [13]:
response2.headers["Content-Type"]

'application/zip'

In [14]:
dataset_date = data["datasetlist"][0]["dataset_date"]

In [25]:
open(f"legiscan_getDatasetRaw_data/getDatasetRaw{dataset_date}.zip", "wb").write(response2.content)
print(f"File saved to legiscan_getDatasetRaw_data/getDatasetRaw{dataset_date}.zip")

FileNotFoundError: [Errno 2] No such file or directory: 'legiscan_getDatasetRaw_data/getDatasetRaw2026-09-13.zip'